src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -75,142 +66,124 @@
      */
     private final int dimensionId;
 
-    /**
-     * List of fields of the colony.
-     */
-    private final List<BlockPos> fields = new ArrayList<>();
-
     /**
      * List of waypoints of the colony.
      */
     private final Map<BlockPos, IBlockState> wayPoints = new HashMap<>();
-    /**
-     * List of achievements within the colony.
-     */
-    @NotNull
-    private final List<Advancement> colonyAchievements;
+
     /**
      * Work Manager of the colony (Request System).
      */
     private final WorkManager workManager = new WorkManager(this);
+
     /**
-     * List of building in the colony.
+     * Building manager of the colony.
      */
-    @NotNull
-    private final Map<BlockPos, AbstractBuilding> buildings = new HashMap<>();
+    private final IBuildingManager buildingManager = new BuildingManager();
+
     /**
-     * List of citizens.
+     * Building manager of the colony.
      */
-    @NotNull
-    private final Map<Integer, CitizenData> citizens = new HashMap<>();
+    private final ICitizenManager citizenManager = new CitizenManager();
+
+    /**
+     * Building manager of the colony.
+     */
+    private final IStatisticAchievementManager statsManager = new StatisticAchievementManager();
+
     /**
      * The Positions which players can freely interact.
      */
     private final Set<BlockPos> freePositions = new HashSet<>();
+
     /**
      * The Blocks which players can freely interact with.
      */
     private final Set<Block> freeBlocks = new HashSet<>();
+
     /**
      * Colony permission event handler.
      */
     private final ColonyPermissionEventHandler eventHandler;
+
     /**
      * Whether there will be a raid in this colony tonight.
      */
     private boolean willRaidTonight = false;
```

Comment:
While you are added a BarbarianManager would be cool

Reply:
Sounds good